catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -0,0 +1,513 @@
+                versionContentItems(updateStorageResponse.getUpdatedContentItems(),
+                        tmpContentPaths,
+                        MetacardVersion.Action.UPDATED_CONTENT);
+
+        List<Metacard> historyMetacards = versionedContentItems.stream()
+                .map(ContentItem::getMetacard)
+                .distinct()
+                .collect(Collectors.toList());
+
+        // These are being staged for delayed execution, to be ran after the storage/catalog has
+        // successfully committed/updated
+        preStaged.add(() -> {
+            CreateStorageRequestImpl createStorageRequest = new CreateStorageRequestImpl(
+                    versionedContentItems,
+                    new HashMap<>());
+            setSkipFlag(createStorageRequest);
+            CreateStorageResponse createStorageResponseResult = getSystemSubject().execute(() -> {
+                CreateStorageResponse createStorageResponse = storageProvider().create(
+                        createStorageRequest);
+                storageProvider().commit(createStorageRequest);
+                overwriteResourceUris(createStorageResponse);
+                return createStorageResponse;
+            });
+
+            return Optional.ofNullable(createStorageResponseResult)
+                    .map(CreateStorageResponse::getProcessingErrors)
+                    .map(Set::isEmpty)
+                    .orElse(false);
+        });
+
+        preStaged.add(() -> {
+            Map<String, Serializable> props = new HashMap<>();
+            props.put(SKIP_VERSIONING, true);
+            CreateResponse createResponse =
+                    getSystemSubject().execute(() -> catalogProvider().create(new CreateRequestImpl(
+                            historyMetacards,
+                            props)));
+            return Optional.ofNullable(createResponse)
+                    .map(CreateResponse::getProcessingErrors)
+                    .map(Set::isEmpty)
+                    .orElse(false);
+        });
+
+        return Optional.of(transactionKey);
+    }
+
+    /**
+     * Commits a set of staged operations.
+     *
+     * @param historianTransactionKey The key associated with a set of operations to commit
+     * @return A List of exceptions, if any, generated by the commit operations
+     */
+    public List<Exception> commit(String historianTransactionKey) {
+        List<Callable<Boolean>> ops = staged.remove(historianTransactionKey);
+        if (ops == null) {
+            LOGGER.warn("There was no operations staged for historian transaction key [{}]",
+                    historianTransactionKey);
+            return null;
+        }
+
+        ArrayList<Exception> exceptions = new ArrayList<>();
+        for (Callable<Boolean> op : ops) {
+            try {
+                op.call();
+            } catch (Exception e) {
+                exceptions.add(e);
+            }
+        }
+        return exceptions;
```

Comment:
Purely for the caller to deal with. Currently the only caller (catalog framework) is not returning those exceptions.